tasks/sep/base-003-fp-granite-prestate/SignFromJson.s.sol

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {SignFromJson as OriginalSignFromJson} from "script/SignFromJson.s.sol";
+import {console2 as console} from "forge-std/console2.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+import {Vm, VmSafe} from "forge-std/Vm.sol";
+import {GnosisSafe} from "safe-contracts/GnosisSafe.sol";
+import {LibString} from "solady/utils/LibString.sol";
+import "@eth-optimism-bedrock/src/dispute/lib/Types.sol";
+import {DisputeGameFactory} from "@eth-optimism-bedrock/src/dispute/DisputeGameFactory.sol";
+import {FaultDisputeGame} from "@eth-optimism-bedrock/src/dispute/FaultDisputeGame.sol";
+import {PermissionedDisputeGame} from "@eth-optimism-bedrock/src/dispute/PermissionedDisputeGame.sol";
+
+contract SignFromJson is OriginalSignFromJson {
+    using LibString for string;
+
+    // Chains for this task.
+    string constant l1ChainName = "sepolia";
+    string constant l2ChainName = "base";
+
+    // Safe contract for this task.
+    GnosisSafe ownerSafe = GnosisSafe(payable(vm.envAddress("OWNER_SAFE")));
+
+    // Known EOAs to exclude from safety checks.
+    address constant l2OutputOracleProposer =
+        0x20044a0d104E9e788A0C984A2B7eAe615afD046b; // cast call $L2OO "PROPOSER()(address)"
+    address constant l2OutputOracleChallenger =
+        0xDa3037Ff70Ac92CD867c683BD807e5A484857405; // In registry addresses.
+    address constant systemConfigOwner =
+        0x0fe884546476dDd290eC46318785046ef68a0BA9; // In registry addresses.
+    address constant batchSenderAddress =
+        0x6CDEbe940BC0F26850285cacA097C11c33103E47; // In registry genesis-system-configs
+    address constant p2pSequencerAddress =
+        0xb830b99c95Ea32300039624Cb567d324D4b1D83C; // cast call $SystemConfig "unsafeBlockSigner()(address)"
+    address constant batchInboxAddress =
+        0xfF00000000000000000000000000000000084532; // In registry yaml.
+    address constant currentPDGProposer =
+        0x727D7c7fCa14b7F3C49a1C816b42a41fe2F709F9;
+    address constant currentPDGChallenger =
+        0x8b8c52B04A38f10515C52670fcb23f3C4C44474F;
+
+    // Currenet dispute game implementations
+    FaultDisputeGame currentFDG;
+    PermissionedDisputeGame currentPDG;
+
+    // New dispute game implementations
+    // https://sepolia.etherscan.io/address/0x48F9F3190b7B5231cBf2aD1A1315AF7f6A554020#code
```

Comment:
New implementation prestate points to the correct one: https://sepolia.etherscan.io/address/0x48F9F3190b7B5231cBf2aD1A1315AF7f6A554020#readContract#F1